packages/loaders/src/Spinner.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  STROKE_WIDTH_FRAMES,
+  DASHARRAY_FRAMES,
+  ROTATION_FRAMES
+} from './utils/spinner-coordinates';
+import { SpinnerCircle, StyledSVG } from './styled-elements';
+import ScheduleContainer from './containers/ScheduleContainer';
+
+const COMPONENT_ID = 'loaders.spinner';
+
+export default class Spinner extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.strokeWidthValues = this.computeFrames(STROKE_WIDTH_FRAMES);
+    this.rotationValues = this.computeFrames(ROTATION_FRAMES);
+    this.dasharrayValues = this.computeFrames(DASHARRAY_FRAMES);
+  }
+
+  static propTypes = {
+    /**
+     * Size of the loader. Can inherit from `font-size` styling.
+     **/
+    size: PropTypes.any,
+    /**
+     * Duration (ms) of the animation. Default is 1250ms.
+     **/
+    duration: PropTypes.number,
+    /**
+     * Color of the loader. Can inherit from `color` styling.
+     **/
+    color: PropTypes.string,
+    /**
+     * Delay in MS to begin loader rendering. This helps prevent
+     * quick flashes of the loader during normal loading times.
+     **/
+    delayMS: PropTypes.number
+  };
+
+  static defaultProps = {
+    size: 'inherit',
+    color: 'inherit',
+    delayMS: 750,
+    duration: 1250
+  };
+
+  state = {
+    frame: 0,
+    rawFrame: 0,
+    totalFrames: 100,
+    delayComplete: false,
+    timestamp: 0
+  };
+
+  computeFrames = frames => {
+    const { duration } = this.props;
+    const { totalFrames } = this.state;
+
+    return Object.entries(frames).reduce((acc, item, index, arr) => {
+      const [frame, value] = item;
+      const [nextFrame, nextValue] = arr[index + 1] || ['100', arr[0][1]];
+      const diff = nextFrame - frame - 1;
```

Comment:
I think this should be `const diff = nextFrame - frame;`. Currently, `computeFrames` produces out-of-order values in the array.